From 1d82246849d43146ee65ebc20abd64e8f931e816 Mon Sep 17 00:00:00 2001
Message-Id: <1d82246849d43146ee65ebc20abd64e8f931e816.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Rafael Barbalho <rafael.barbalho@intel.com>
Date: Wed, 30 Jul 2014 17:31:06 +0100
Subject: [PATCH 212/312] FOR_UPSTREAM [VPG]: drm/i915/chv: Don't access
 non-existant PWM registers

Only pipes A & B have PWM control registers.

Change-Id: I3c93d7c55af91df905907fd2015ef2ee29c834bb
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>

Conflicts:
	drivers/gpu/drm/i915/intel_panel.c
---
 drivers/gpu/drm/i915/intel_panel.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 1f6a80a..0cca4ea 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -1252,7 +1252,12 @@ static int vlv_setup_backlight(struct intel_connector *connector)
 	u32 ctl, ctl2, val;
 
 	for_each_pipe(pipe) {
-		u32 cur_val = I915_READ(VLV_BLC_PWM_CTL(pipe));
+		u32 cur_val;
+
+		if (pipe >= PIPE_C)
+			continue;
+
+		cur_val = I915_READ(VLV_BLC_PWM_CTL(pipe));
 
 		/* Skip if the modulation freq is already set */
 		if (cur_val & ~BACKLIGHT_DUTY_CYCLE_MASK)
-- 
1.7.9.5

